File: frontend/src/microservices/content/objects/CheckReviewEligibilityPlannerRequest.ts
// CheckReviewEligibilityPlannerRequest：content 域评论资格判断请求对象。

export type CheckReviewEligibilityPlannerRequest = {
  userId: string
  orderItemId: string
}

export const checkReviewEligibilityPlannerRequestFromJson = (json: string): CheckReviewEligibilityPlannerRequest =>
  JSON.parse(json) as CheckReviewEligibilityPlannerRequest

export const checkReviewEligibilityPlannerRequestToJson = (value: CheckReviewEligibilityPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/content/objects/CheckReviewEligibilityPlannerRequest.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CheckReviewEligibilityPlannerRequest(
    userId: String,
    orderItemId: String
)
object CheckReviewEligibilityPlannerRequest:
  given sourceEncoder: Encoder[CheckReviewEligibilityPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CheckReviewEligibilityPlannerRequest] = deriveDecoder
